### PR TITLE
Fix bug with Option of zero or false

### PR DESCRIPTION
**Pull request by igorkamyshev**
Hello! Your library is so great. But, I found one minor bug.
```js
Option.of(false).getOrElse(true) === true
```
but, must be `false`. Same thing with `Option.of(0)`

I fixed it and wrote tests. Please, review my PR and give me your feedback.

### CODE REVIEW

**Comment by Uncleseneca** [comment]
Great lib!

File: test.ts [hunk 1]
```diff
@@ -47,6 +47,8 @@ test('Some#flatMap', t => t.is(Option.of(null).flatMap(() => Option.of(2)).isEmp
 test('Some#get', t => t.is(Option.of(3).get(), 3))
 
 test('Some#getOrElse', t => t.is(Option.of(1).getOrElse(3), 1))
+test('Some(0)#getOrElse', t => t.is(Option.of(0).getOrElse(12), 0))
+test('Some(false)#getOrElse', t => t.is(Option.of(false).getOrElse(true), false))
```
Comment (bcherny) [inline test.ts:51]: Would you mind adding a test for `NaN` and `""` too, while you're at it?

**Review by bcherny — CHANGES_REQUESTED** [review]
Really great catch @igorkamyshev!

File: test.ts (hunk 1 above)
Comment (igorkamyshev) [reply test.ts:51]: Sure, done.

**Review by bcherny — APPROVED** [review]
Looks great. Thanks for the contribution @igorkamyshev!

**Comment by bcherny** [comment]
Published `0.7.0`

**Pull request merged**